### ISSUE TITLE
Change Backbone to Swin Transformer

### ISSUE THREAD

**Issue by Soar-Sir** [damo-cv/TransReID#46]
Hello, first of all thank you guys for your excellent work. Have you ever tried to replace the backbone network of TransReID with Swin Transformer, because Swin seems to surpass ViT and DeiT in various fields of computer vision. I don’t know if Swin is in the re-identification field will also perform better. I look forward to your reply, thank you.

**Comment by michuanhaohao** [comment]
Hi.

Swin transformer performs well in person ReID. There are some results:
<div class="lake-content" typography="classic">

Model | Market | Duke
-- | -- | --
Vit-base (Imagenet21k pre-training) | 86.8/94.4 | 78.9/89.3
Swin-base (Imagenet21k pre-training) | 89.3/95.4 | 81.2/90.4
</div>

**Comment by Soar-Sir** [comment]
Thanks! I want to know what codes you have changed？After I changed Backbone to swin-transformer, at the same time, I changed some codes in make_model.py, but the improvement effect is not obvious. Maybe my parameters are not set  properly? If possible, can you share your files? It will be very helpful to me. Thanks again.

**Comment by michuanhaohao** [comment]
I only changed the window size for some special resolutions. You can try 224*224 resolution at first., which needs no change. The pre-trained models are pre-trained on ImageNet21K.

```
def pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate):
    if img_size in ([256,128],[256,256],[384,128]):
        window_size = 8
    elif img_size in ([224,224],[224,112]):
        window_size =7
    elif img_size in ([384,384],[384,192]):
        window_size = 12
    elif img_size in ([192,192],):
        window_size = 6
    else:
        print('Window size dose not match!')
    print('Window size is set to %d'%window_size)
    print('using drop_out rate is : {}'.format(drop_rate))
    print('using attn_drop_out rate is : {}'.format(attn_drop_rate))
    print('using drop_path rate is : {}'.format(drop_path_rate))
    return window_size

def swin_base_patch4_window7_224(img_size=224,drop_rate=0.0, attn_drop_rate=0.0, drop_path_rate=0.1,camera_num=0, view_num=0, **kwargs):
    window_size = pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate)
    model = SwinTransformer(img_size = img_size, patch_size=4, window_size=window_size, embed_dim=128, depths=(2, 2, 18, 2), num_heads=(4, 8, 16, 32), drop_path_rate=drop_path_rate, drop_rate=drop_rate, attn_drop_rate=attn_drop_rate, **kwargs)
    return model

def swin_small_patch4_window7_224(img_size=224,drop_rate=0.0, attn_drop_rate=0.0, drop_path_rate=0.1,camera_num=0, view_num=0, **kwargs):
    window_size = pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate)
    model = SwinTransformer(img_size = img_size, patch_size=4, window_size=window_size, embed_dim=96, depths=(2, 2, 18, 2), num_heads=(3, 6, 12, 24), drop_path_rate=drop_path_rate, drop_rate=drop_rate, attn_drop_rate=attn_drop_rate, **kwargs)
    return model
```

**Comment by sssqfaa** [comment]
Hello, I modified the code according to the method you said, but it still failed to reach the result you showed. I suspect it may be related to the learning rate. Could you please tell me the learning rate and batchsize you have used？

**Comment by Soar-Sir** [comment]
> Hello, I modified the code according to the method you said, but it still failed to reach the result you showed. I suspect it may be related to the learning rate. Could you please tell me the learning rate and batchsize you have used？

I also encountered the same situation, the accuracy of the model is not as high as expected.

**Comment by CarrieYpi** [comment]
> I only changed the window size for some special resolutions. You can try 224*224 resolution at first., which needs no change. The pre-trained models are pre-trained on ImageNet21K.
> 
> ```
> def pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate):
>     if img_size in ([256,128],[256,256],[384,128]):
>         window_size = 8
>     elif img_size in ([224,224],[224,112]):
>         window_size =7
>     elif img_size in ([384,384],[384,192]):
>         window_size = 12
>     elif img_size in ([192,192],):
>         window_size = 6
>     else:
>         print('Window size dose not match!')
>     print('Window size is set to %d'%window_size)
>     print('using drop_out rate is : {}'.format(drop_rate))
>     print('using attn_drop_out rate is : {}'.format(attn_drop_rate))
>     print('using drop_path rate is : {}'.format(drop_path_rate))
>     return window_size
> 
> def swin_base_patch4_window7_224(img_size=224,drop_rate=0.0, attn_drop_rate=0.0, drop_path_rate=0.1,camera_num=0, view_num=0, **kwargs):
>     window_size = pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate)
>     model = SwinTransformer(img_size = img_size, patch_size=4, window_size=window_size, embed_dim=128, depths=(2, 2, 18, 2), num_heads=(4, 8, 16, 32), drop_path_rate=drop_path_rate, drop_rate=drop_rate, attn_drop_rate=attn_drop_rate, **kwargs)
>     return model
> 
> def swin_small_patch4_window7_224(img_size=224,drop_rate=0.0, attn_drop_rate=0.0, drop_path_rate=0.1,camera_num=0, view_num=0, **kwargs):
>     window_size = pre_settings(img_size, drop_rate, attn_drop_rate, drop_path_rate)
>     model = SwinTransformer(img_size = img_size, patch_size=4, window_size=window_size, embed_dim=96, depths=(2, 2, 18, 2), num_heads=(3, 6, 12, 24), drop_path_rate=drop_path_rate, drop_rate=drop_rate, attn_drop_rate=attn_drop_rate, **kwargs)
>     return model
> ```

我也在更换为swin-transformer时,遇见了问题，代码中我是加载了预训练模型但似乎没有加载上，效果非常差，请问您能够分享更改为swin transformer完整的backbone代码以及make_model中的代码吗，非常感谢！！
